packages/@ember/-internals/container/lib/container.ts

```diff
@@ -485,7 +485,7 @@ class FactoryManager<T, C> {
     this.injections = undefined;
     setFactoryFor(this, this);
 
-    if (factory) {
+    if (isInstantiatable(container, fullName)) {
```

Comment:
Do we also need to check if `factory` is non-falsy? (The prior if was guarding for when factory was undefined but the replacement doesn’t).

Reply:
I ran test without any guard and they failed. When I added the new guard they passed again. So the new guard also covers the test cases of the old guard but also prevents static factories from getting the association.